### PR TITLE
Adding documentation for running workloads

### LINKED ISSUE #383: [Update Documentation] Getting started with OpenSearch-Benchmark!
**Context:**
There are different ways to get started with running a cluster and running a benchmark against it. 
I was trying to use the docker-compose.yml from [opensearch.org](http://opensearch.org/), but was facing SSL issues, so I thought I'd make this "guide" so everyone can get started with it quickly!
Below, I'm going to outline the simplest set of instructions I've found for getting started with OSB for the first time. Huge thanks to @IanHoang and @rishabh6788 for helping everyone in Slack. 

**Steps:**
This can be reproduced on both Linux and Windows (WSL). 
1. Download the tar.gz from https://opensearch.org/downloads.html. If you are using WSL, I'd recommend downloading this using wget into your /home/<user> directory instead of /mnt/c. 
```
wget https://artifacts.opensearch.org/releases/bundle/opensearch/2.10.0/opensearch-2.10.0-linux-x64.tar.gz
```
2. Extract the folder
```
tar -xf opensearch-2.10.0-linux-x64.tar.gz
cd opensearch-2.10.0
```
3. Add following settings to the opensearch.yml file under the config directory 
```
vim config/opensearch.yml
```
```
discovery.type: single-node 
plugins.security.disabled: true 
```
4. Run the opensearch-tar-install.sh script.
```
bash opensearch-tar-install.sh
```
5. Check the output of `curl.exe "http://localhost:9200/_cluster/health?pretty"`. Output should be similar to this:
```
{
  "cluster_name" : "<name>",
  "status" : "green",
  "timed_out" : false,
  "number_of_nodes" : 1,
  "number_of_data_nodes" : 1,
  "discovered_master" : true,
  "discovered_cluster_manager" : true,
  "active_primary_shards" : 3,
  "active_shards" : 3,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 0,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 100.0
}
```
6. Before we move on, make sure to have gone through the DEVELOPER_GUIDE.md and activated your virtual environment. TIP: Just to avoid Pyenv breaking, run the following command before you reach this step ^^
```
sudo apt-get install -y make build-essential libssl-dev zlib1g-dev libbz2-dev \
libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev \
xz-utils tk-dev libffi-dev liblzma-dev git
```
7. For the last step, we'd be executing the geonames benchmark which can be found from the workloads repo. 
```
opensearch-benchmark execute-test --pipeline=benchmark-only --workload=geonames --target-host=127.0.0.1:9200 --test-mode --workload-params '{"number_of_shards":"1","number_of_replicas":"0"}'
```

And we should be done! You should be seeing the performance metrics! 
**If you are not seeing any results immediately, it should be an indicator that there is an issue with your cluster**. Use the command below to view the cluster logs 
```
tail -f ~/.bencmark/logs/bechmark.log
```

### PR DESCRIPTION
**Pull request by AkshathRaghav**
### Description
[Describe what this change achieves]Adding documentation for running workloads

### Issues Resolved
#383 

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -24,6 +24,13 @@ This document will walk you through on what's needed to start contributing code
   - **Pyenv** : Install `pyenv` and follow the instructions in the output of `pyenv init` to set up your shell and restart it before proceeding.
     For more details please refer to the [PyEnv installation instructions](https://github.com/pyenv/pyenv#installation).
 
+    Install the following modules to continue with the next steps:
```
Comment (IanHoang) [inline DEVELOPER_GUIDE.md:27]: Is this specific to testing on a debian-based distro? If so, maybe we should include this at the end as optional based on OS system.

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -74,6 +83,74 @@ This is typically created in PyCharm IDE by visiting the `Python Interpreter`, s
 `
 In order to run tests within the PyCharm IDE, ensure the `Python Integrated Tools` / `Testing` / `Default Test Runner` is set to `pytest`.
 
+## Running Workloads
```
Comment (IanHoang) [inline DEVELOPER_GUIDE.md:86]: Nit: `Running Workloads` --> `Setting Up an Local OpenSearch Cluster`

This title should be more descriptive for a couple reasons:
- We already have documentation on the official website that details how to run workloads, so this might confuse users
- Most users seem to be using OSB to benchmark larger and external clusters instead of performance testing their local computer. 

Maybe this section could include a brief explanation why developers might want to setup a local cluster (e.g. for simple feature testing or end to end testing etc.)

File: DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -74,6 +83,74 @@ This is typically created in PyCharm IDE by visiting the `Python Interpreter`, s
...
+
+### Installation
```
Comment (IanHoang) [inline DEVELOPER_GUIDE.md:88]: I'd rephrase this to `OpenSearch Installation`, just cause this developer guide pertains to OSB and we already have an "Installation" section

File: DEVELOPER_GUIDE.md [hunk 4]
```diff
@@ -74,6 +83,74 @@ This is typically created in PyCharm IDE by visiting the `Python Interpreter`, s
...
+
+Download the latest release of OpenSearch from https://opensearch.org/downloads.html. If you are using WSL, make sure to download it into your `/home/<user>` directory instead of `/mnt/c`. 
+```
+wget https://artifacts.opensearch.org/releases/bundle/opensearch/<x.x.x>/opensearch-<x.x.x>-linux-x64.tar.gz
+tar -xf opensearch-x.x.x-linux-x64.tar.gz
+cd opensearch-x.x.x
+```
+NOTE: Have Docker running in the background for the next steps. Refer to the installation instructions [here](https://docs.docker.com/compose/install/).
+
+### Setup
```
Comment (IanHoang) [inline DEVELOPER_GUIDE.md:98]: Would be more descriptive here to differentiate that this is for OpenSearch setup and not OSB

File: DEVELOPER_GUIDE.md [hunk 5]
```diff
@@ -74,6 +83,74 @@ This is typically created in PyCharm IDE by visiting the `Python Interpreter`, s
...
+  "delayed_unassigned_shards" : 0,
+  "number_of_pending_tasks" : 0,
+  "number_of_in_flight_fetch" : 0,
+  "task_max_waiting_in_queue_millis" : 0,
+  "active_shards_percent_as_number" : 100.0
+}
+```
+Now, you have a local cluster running! You can connect to this and run the workload for the next step. 
+
+### Running the workload
```
Comment (IanHoang) [inline DEVELOPER_GUIDE.md:138]: Would be more descriptive so that this is for running an end to end test / workload on a locally installed OpenSearch cluster

**Comment by IanHoang** [comment]
For any new sections added, could you incorporate them into the table of contents?

**Review by IanHoang — CHANGES_REQUESTED** [review]
Left a few comments!

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (AkshathRaghav) [reply DEVELOPER_GUIDE.md:27]: Got it. Will add!

**Review by IanHoang — APPROVED** [review]

File: DEVELOPER_GUIDE.md [hunk 6]
```diff
@@ -74,6 +83,74 @@ This is typically created in PyCharm IDE by visiting the `Python Interpreter`, s
 `
 In order to run tests within the PyCharm IDE, ensure the `Python Integrated Tools` / `Testing` / `Default Test Runner` is set to `pytest`.
 
+## Setting Up an Local OpenSearch Cluster For OSB Development (Optional)
```
Comment (IanHoang) [inline DEVELOPER_GUIDE.md:86]: Could this section be added to the table of contents @AkshathRaghav ?

File: DEVELOPER_GUIDE.md [hunk 7]
```diff
@@ -38,7 +46,9 @@ This document will walk you through on what's needed to start contributing code
 
 ### Setup
 
-To develop OSB properly, it is recommended that you fork the official OpenSearch Benchmark repository.
+To develop OSB properly, it is recommended that you fork the official OpenSearch Benchmark repository. 
+
+For those working on WSL2, it is recommended to clone the repository and set up the working environment within the Linux subsystem. Refer to the guide for setting up WSL2 on [Visual Studio Code](https://code.visualstudio.com/docs/remote/wsl) or [PyCharm](https://www.jetbrains.com/help/pycharm/using-wsl-as-a-remote-interpreter.html#create-wsl-interpreter).
```
Comment (gkamat) [inline DEVELOPER_GUIDE.md:51]: It would be good to indicate that WSL is a supported platform for development, if that is the case.  Some documentation on whether it is WSL1 or WSL2 (or both) would be helpful.

File: DEVELOPER_GUIDE.md [hunk 8]
```diff
@@ -74,6 +84,74 @@ This is typically created in PyCharm IDE by visiting the `Python Interpreter`, s
...
+## Setting Up a Local OpenSearch Cluster For OSB Development (Optional)
+
+### OpenSearch Installation  
+
+Download the latest release of OpenSearch from https://opensearch.org/downloads.html. If you are using WSL, make sure to download it into your `/home/<user>` directory instead of `/mnt/c`. 
+```
+wget https://artifacts.opensearch.org/releases/bundle/opensearch/<x.x.x>/opensearch-<x.x.x>-linux-x64.tar.gz
+tar -xf opensearch-x.x.x-linux-x64.tar.gz
+cd opensearch-x.x.x
+```
```
Comment (gkamat) [inline DEVELOPER_GUIDE.md:96]: Some of this material is being updated in the doc repo https://github.com/opensearch-project/documentation-website.  It would be good to consolidate the two and direct readers there.

File: DEVELOPER_GUIDE.md [hunk 9]
```diff
@@ -74,6 +84,74 @@ This is typically created in PyCharm IDE by visiting the `Python Interpreter`, s
...
+### OpenSearch Cluster setup
+
+Add the following settings to the `opensearch.yml` file under the config directory 
+```
+vim config/opensearch.yml
+```
+```
+#
+discovery.type: single-node 
+plugins.security.disabled: true
```
Comment (gkamat) [inline DEVELOPER_GUIDE.md:108]: This should not be necessary, if certificate checking is eliminated. See https://github.com/opensearch-project/documentation-website/pull/5177.

File: DEVELOPER_GUIDE.md [hunk 10]
```diff
@@ -100,7 +178,6 @@ Invoke integration tests by running the following command within the root direct
 make it
 ```
 
-Integration tests are expected to run for approximately 20-30 mins.
```
Comment (gkamat) [inline DEVELOPER_GUIDE.md:103]: Why was this line removed?  Is the duration different for WSL, perhaps?

**Review by gkamat — CHANGES_REQUESTED** [review]

File: DEVELOPER_GUIDE.md (hunk 10 above)
Comment (AkshathRaghav) [reply DEVELOPER_GUIDE.md:103]: I placed it in the line above and bolded the timeframe there. I did this because I had not seen it the first time and I was confused as to why it was taking so long.

File: DEVELOPER_GUIDE.md (hunk 7 above)
Comment (AkshathRaghav) [reply DEVELOPER_GUIDE.md:51]: As far as I have tested, it is possible to work on both. Shall I just replace WSL2 with WSL in the documentation, or explicitly say that both are ok to work with.

File: DEVELOPER_GUIDE.md (hunk 8 above)
Comment (AkshathRaghav) [reply DEVELOPER_GUIDE.md:96]: I just saw your [PR](https://github.com/opensearch-project/documentation-website/pull/5177). Shall I remove this section and just redirect to that page once the PR is merged?

Comment (IanHoang) [reply DEVELOPER_GUIDE.md:96]: @AkshathRaghav Yes, let's remove this section and redirect it to the page once that PR is merged in. We will have the other PR merged in shortly.

File: DEVELOPER_GUIDE.md (hunk 7 above)
Comment (IanHoang) [reply DEVELOPER_GUIDE.md:51]: I recommend explicitly saying that both WSL and WSL2 work with OSB

**Review by gkamat — APPROVED** [review]

**Comment by gkamat** [comment]
Any additional changes can be added in the next iteration.
